### ISSUE TITLE
Dynamic Fork task limit? 4,000 tasks seems to cause CPU issue

### ISSUE THREAD

**Issue by rickfish** [Netflix/conductor#1767]
We have a workflow that has 4,000 SUB_WORKFLOW tasks in a FORK_JOIN_DYNAMIC task. When that workflow executes, the CPU spikes on our Conductor instance and it causes lots of problems including Conductor not starting, probably because the background sweeper thread has an issue processing it.

Has anyone noticed this behavior? Is there a recommended maximum for the number of tasks that can be in a dynamic fork task? If this is a known issue, has anyone come up with a recommended solution that doesn't involve creating 4,000 individual workflows and having some sort of aggregator to join them together and get stats when they all have completed?

**Comment by apanicker-nflx** [comment]
We have experimented with 1000 SUB_WORKFLOW tasks within a FORK_JOIN_DYNAMIC with successful executions. However, we did notice the CPU spike. We believe the cause for this is that during a workflow evaluation all tasks within the workflow are loaded into memory and the state machine runs evaluation for each task. 
Given the current architecture, this would mean overhauling the state machine evaluation logic completely to allow such a huge number of tasks. We are working on improving this design for the next iteration of Conductor (3.0).
